composed-message-processor/composed-message-processor.bal

```diff
@@ -0,0 +1,55 @@
+import ballerina/http;
+
+type SalesByStateRequest record {|
+    string[] states;
+|};
+
+type SalesByState record {|
+    decimal revenue;
+    decimal operatingExpenses;
+    int production;
+    int totalEmployees;
+|};
+
+type AggregratedSales record {|
+    map<decimal> revenueByState = {};
+    decimal totalRevenue = 0.0;
+    decimal maxRevenue = 0.0;
+    string maxRevenueState = "";
+    map<decimal> operatingExpensesByState = {};
+    decimal totalOperatingExpenses = 0.0;
+    int totalProduction = 0;
+    map<int> productivityByState = {};
+|};
+
+final map<http:Client> stateRoutes = {
+    Texas: check new ("http://api.texas.office.com.balmock.io"),
+    Ohio: check new ("http://api.ohio.office.com.balmock.io"),
+    Florida: check new ("http://api.florida.office.com.balmock.io")
+};
+
+service /api/v1 on new http:Listener(8080) {
+    resource function post sales\-dashboard(SalesByStateRequest salesRequest) returns AggregratedSales|error {
+        AggregratedSales summary = {};
+        foreach string state in salesRequest.states {
+            http:Client? stateClient = stateRoutes[state];
+            if stateClient == () {
+                return error("Invalid state provided");
+            }
+            SalesByState salesByState = check stateClient->/sales();
+            aggregateAllSales(summary, state, salesByState);
+        }
+        return summary;
+    }
+}
+
+function aggregateAllSales(AggregratedSales summary, string state, SalesByState salesByState) {
```

Comment:
Functions doesn't accept an array of SalesByState, so All doesn't make since. I think `aggregateSales` is better.